### PR TITLE
add func test for reset after timeout

### PR DESCRIPTION
**Pull request by ychebotarev**
## What changed?

Add a functional test for "reset after timeout" scenario.

## Why?

If was reported that this bug still exist. I think we need to have a functional test to cover this

## How did you test it?

In this PR only test changed.

## Potential risks
No risk

### CODE REVIEW

File: tests/reset_workflow.go [hunk 1]
```diff
@@ -216,6 +218,132 @@ func (s *FunctionalSuite) TestResetWorkflow() {
...
+	we, err0 := s.client.StartWorkflowExecution(NewContext(), request)
+	s.NoError(err0)
+
+	s.Logger.Info("StartWorkflowExecution", tag.WorkflowRunID(we.RunId))
+
+	time.Sleep(time.Second) //nolint:forbidigo
+
+	var historyEvents []*historypb.HistoryEvent
+GetHistoryLoop:
+	for i := 0; i < 10; i++ {
```
Comment (yycptt) [inline tests/reset_workflow.go:248]: use Eventually?

File: tests/reset_workflow.go [hunk 2]
```diff
@@ -216,6 +218,132 @@ func (s *FunctionalSuite) TestResetWorkflow() {
...
+			StartTimeFilter: startFilter,
+			Filters: &workflowservice.ListClosedWorkflowExecutionsRequest_ExecutionFilter{ExecutionFilter: &filterpb.WorkflowExecutionFilter{
+				WorkflowId: tv.WorkflowID(),
+			}},
+		})
+		s.NoError(err3)
+		closedCount = len(resp.Executions)
+		if closedCount == 0 {
+			s.Logger.Info("Closed WorkflowExecution is not yet visible")
+			time.Sleep(500 * time.Millisecond) //nolint:forbidigo
```
Comment (yycptt) [inline tests/reset_workflow.go:288]: ditto

**Review by yycptt — APPROVED** [review]

File: tests/reset_workflow.go (hunk 1 above)
Comment (ychebotarev) [reply tests/reset_workflow.go]: done

File: tests/reset_workflow.go (hunk 2 above)
Comment (ychebotarev) [reply tests/reset_workflow.go]: changed to eventually as well.

**Pull request merged**